app/src/main/java/com/example/househomey/AddItemFragment.java

```diff
@@ -51,39 +60,82 @@ public AddItemFragment(CollectionReference itemRef) {
+        TextWatcher emptyTextWatcher = new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+            }
+
+            @Override
+            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+            }
+
+            @Override
+            public void afterTextChanged(Editable editable) {
+                if (editable.equals(descView.getEditableText())) {
+                    isRequiredFieldEmpty(R.id.add_item_description_layout);
+                } else if (editable == costView.getEditableText()) {
+                    isRequiredFieldEmpty(R.id.add_item_cost_layout);
+                }
+            }
+        };
+        descView.addTextChangedListener(emptyTextWatcher);
+        costView.addTextChangedListener(emptyTextWatcher);
 
         // Add listener for confirm and back buttons
         rootView.findViewById(R.id.add_item_confirm_button).setOnClickListener(v -> addItem());
-        rootView.findViewById(R.id.add_item_back_button).setOnClickListener(v -> returnToHomeScreen());
+        rootView.findViewById(R.id.add_item_back_button).setOnClickListener(v -> navigateToFragmentPage((AppCompatActivity) getContext(), new HomeFragment(itemRef)));
 
         return rootView;
     }
 
     /**
-     * This shows a datePicker for the user to select the date of the item acquisition
+     * Checks whether the text input for a required form field is empty.
+     * Additionally, if empty, it sets an inline error message on the input field.
+     * If not empty, it removes the inline error.
+     *
+     * @param id Id of the required field's TextInputLayout
+     * @return a boolean indicating if the field is empty
      */
-    private void showDatePicker() {
-        MaterialDatePicker<Long> datePicker = MaterialDatePicker.Builder.datePicker()
-                .setInputMode(MaterialDatePicker.INPUT_MODE_CALENDAR)
-                .build();
-
-        datePicker.addOnPositiveButtonClickListener(selection -> {
-            dateAcquired = new Date(selection);
-            dateTextView.setText(datePicker.getHeaderText());
-        });
-
-        datePicker.show(getParentFragmentManager(), "Date Picker");
+    private boolean isRequiredFieldEmpty(int id) {
+        TextInputLayout textInputLayout = (TextInputLayout) getView().findViewById(id);
+        if (TextUtils.isEmpty(textInputLayout.getEditText().getText().toString().trim())) {
+            textInputLayout.setError("This field is required");
+            return true;
+        }
+        textInputLayout.setError(null);
+        return false;
     }
 
     /**
-     * Adds the user input data to the firestore database
+     * Adds the user input data to a new item in their Firestore collection
      */
     private void addItem() {
-        // TODO: add input validation
+        // Check that required fields are filled before submitting
```

Comment:
I wonder if we might be better off shifting this logic to the item class... My thinking is something along these lines:
```
make the map
try {
  Item newItem = new Item(map); // Either need to generate id prior somehow or make a constructor with no id
} catch (NullPointerException) {
  return;
}

itemRef.add(item.data())... // Would need to define a mapping in Item called data()
```
This means we guarantee you can actually make an Item from the info. Kinda like regex checking Date vs actually creating a Date object

Reply:
We could do that approach; do you think it's worth it? Long term, maybe, for photos/tags. Right now, I think that would only really help with verifying that the `estimated value` is valid, because the rest are just strings...

Reply:
I think this way leaves us very prone to errors evidenced by my insertion of bad data that will now screw everyone over cuz of the cache. *Nothing* can go wrong if you can't insert bad data in the db... and if that check is obscurely scattered around a fragment... I think it's worth the effort